replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/TopicCommit.scala

```diff
@@ -1,44 +1,132 @@
+ * ([[TopicConsumer]], [[ConsumeTopic]]).
+ *
+ * @tparam F effect type
+ */
+private[journal] trait TopicCommit[F[_]] {
+// journal-private because the places where it is used are journal-private - TopicConsumer, ConsumeTopic
 
+  /**
+   * Mark offsets for commit.
+   * 
+   * The method should be called after each processing step, i.e., after each consumer poll result processing.
+   * 
+   * Please note that offsets passed here should follow the same logic as for Java consumer commit methods:
+   * if offset N is known to be processed, offset N+1 should be committed - offset from which you want to start your
+   * processing next time.
+   * 
+   * Depending on the underlying implementation, this method might execute commit synchronously with the call or
+   * schedule it for later.
+   * 
+   * @param offsets partition offsets to commit
+   */
   def apply(offsets: Nem[Partition, Offset]): F[Unit]
+
+  /**
+   * Notify topic commit logic about new partitions assigned during rebalance - called in
+   * [[com.evolutiongaming.skafka.consumer.RebalanceListener1.onPartitionsAssigned]].
+   * 
+   * @param partitions assigned partitions
+   *                   
+   * @return [[RebalanceCallback]]
+   */
+  def onPartitionsAssigned(@nowarn("cat=unused") partitions: NonEmptySet[Partition]): RebalanceCallback[F, Unit] =
+    RebalanceCallback.pure(())
+
+  /**
+   * Notify topic commit logic about partitions revoked during rebalance - called in
+   * [[com.evolutiongaming.skafka.consumer.RebalanceListener1.onPartitionsRevoked]].
+   * 
+   * Could be used to commit offsets for revoked partitions using
+   * [[RebalanceCallback.commit]] in order not to lose progress during rebalance.
+   *
+   * @param partitions revoked partitions
+   *
+   * @return [[RebalanceCallback]]
+   */
+  def onPartitionsRevoked(@nowarn("cat=unused") partitions: NonEmptySet[Partition]): RebalanceCallback[F, Unit] =
+    RebalanceCallback.pure(())
+
+  /**
+   * Notify topic commit logic about partitions lost - called in
+   * [[com.evolutiongaming.skafka.consumer.RebalanceListener1.onPartitionsLost]].
+   *
+   * When partitions are lost as opposed to being revoked:
+   * [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener#onPartitionsLost(java.util.Collection)]]
+   * 
+   * Usually nothing could be done in this case apart from removing the lost partitions from internal state, if there
+   * is any.
+   *
+   * @param partitions lost partitions
+   *
+   * @return [[RebalanceCallback]]
+   */
+  def onPartitionsLost(@nowarn("cat=unused") partitions: NonEmptySet[Partition]): RebalanceCallback[F, Unit] =
+    RebalanceCallback.pure(())
 }
 
-object TopicCommit {
+private[journal] object TopicCommit {
 
-  def empty[F[_]: Applicative]: TopicCommit[F] = (_: Nem[Partition, Offset]) => ().pure[F]
+  def empty[F[_]: Applicative]: TopicCommit[F] = new EmptyTopicCommit[F]
 
+  @deprecated("renamed to sync", since = "4.2.0")
```

Comment:
IMHO message should be about "how to deal with it", not "what was done":
```suggestion
  @deprecated("use `sync` instead", since = "4.2.0")
```

Reply:
`use sync instead` might suggest something changed in the behaviour and I need to pay attention. I wanted to emphasize that it is only a renaming change, the behaviour is the same.